app/renderer/components/Session/CapabilityControl.js

```diff
@@ -4,27 +4,33 @@ import SessionStyles from './Session.css';
 import { remote } from 'electron';
 import { FileOutlined } from '@ant-design/icons';
 import { INPUT } from '../AntdTypes';
+import _ from 'lodash';
+import log from 'electron-log';
 
 const {dialog} = remote;
 
-
 export default class NewSessionForm extends Component {
 
-  getLocalFilePath (success) {
-    dialog.showOpenDialog((filepath) => {
-      if (filepath) {
-        success(filepath);
+  async getLocalFilePath () {
+    try {
+      const {canceled, filePaths} = await dialog.showOpenDialog({properties: ['openFile']});
+      if (!canceled && !_.isEmpty(filePaths)) {
+        return filePaths[0];
       }
-    });
-    this.handleSetType = this.handleSetType.bind(this);
+    } catch (e) {
+      log.error(e);
+    }
   }
 
   render () {
     const {cap, onSetCapabilityParam, isEditingDesiredCaps, id, t} = this.props;
 
-    const buttonAfter = <FileOutlined
+    const buttonAfter = (currentPath) => <FileOutlined
       className={SessionStyles['filepath-button']}
-      onClick={() => this.getLocalFilePath((filepath) => onSetCapabilityParam(filepath[0]))} />;
+      onClick={async () => {
+        const filePath = await this.getLocalFilePath();
+        onSetCapabilityParam(filePath ? filePath : currentPath) ;
```

Comment:
could be simplified to `filePath || currentPath`

Reply:
what is the purpose of calling onSetCapabilityParam with `currentPath` argument?

Reply:
To keep the current inputted path when file browsing was cancelled. Previous filepath behaviour (before #1492) also behaved as the same.